### ISSUE TITLE
[Bug]: Can't resolve 'lodash.zip' 

### ISSUE THREAD

**Issue by hikerpig** [web-infra-dev/rspack#7838]
### System Info

  System:
    OS: macOS 14.6.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 134.63 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.15.1 - ~/.local/state/fnm_multishells/98630_1725883461237/bin/node
    Yarn: 1.22.22 - ~/Library/pnpm/yarn
    npm: 9.9.3 - ~/Library/pnpm/npm
    pnpm: 8.13.1 - ~/Library/pnpm/pnpm
  Browsers:
    Chrome: 128.0.6613.120
    Edge: 128.0.2739.67
    Safari: 17.6
  npmPackages:
    @rspack/cli: ^1.0.3 => 1.0.3
    @rspack/core: ^1.0.3 => 1.0.3

### Details

rspack will report error in the line `import zip from 'lodash.zip'`.

This behavior did not occur to me in rspack 1.0.0-beta4,  but when I upgrade to 1.0.0 it complains about not able to resolve `lodash.zip` even though I did install it and can find it in node_modules.

```
> rspack-lodash-zip-repro@1.0.0 build /Volumes/hik/projects/rspack-lodash-zip-repro
> cross-env NODE_ENV=production rspack build

ERROR in ./src/index.js 2:0-29
  × Module not found: Can't resolve 'lodash.zip' in '/Volumes/hik/projects/rspack-lodash-zip-repro/src'
   ╭─[2:16]
 1 │ import './index.css';
 2 │ import zip from 'lodash.zip';
   ·                 ────────────
 3 │ console.log('zip fn is', zip);
 4 │ document.querySelector('#root').innerHTML = `
```

### Reproduce link

https://github.com/hikerpig/rspack-lodash-zip-repro

### Reproduce Steps

1. `pnpm install`
2. `pnpm run build`

**Comment by inottn** [comment]
It seems to be fixed, you can try it after 1.0.4 is released.

**Comment by LingyuCoder** [comment]
1.0.4 has been released
